src/main/java/com/mercuryCyclists/Inventory/service/ProductService.java

```diff
@@ -93,15 +93,27 @@ public boolean addPart(Long productId, Part part) {
         return true;
```

Comment:
In your addPart method, you need to validate that the supplier exists. Same with the update